site/app/models/gradeable/Gradeable.php

```diff
@@ -491,12 +491,42 @@ public function setPeerGradersList($input) {
         else {
             $this->core->getQueries()->clearPeerGradingAssignment($this->getId());
             foreach ($input as $row_num => $vals) {
-                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $this->getId());
+                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $vals["feedback"]);
                 $this->modified = true;
                 $this->peer_grading_pairs = $this->core->getQueries()->getPeerGradingAssignment($this->getId());
             }
         }
     }
+    
+    public function setPeerFeedback($grader_id, $student_id, $feedback_full, $feedback_id) {
+        $bad_input = [];
+        if ($this->core->getQueries()->getUserById($grader_id) == null) {
+            array_push($bad_input, ($grader_id));
+        }
+        if ($this->core->getQueries()->getUserById($student_id) == null) {
+            array_push($bad_input, ($student_id));
+        }
+        if (!empty($bad_input)) {
+            $msg = "The given user id is not valid: ";
+            array_walk($bad_input, function ($val) use (&$msg) {
+                $msg .= " {$val}";
+            });
+            $this->core->addErrorMessage($msg);
+        }
```

Comment:
Will this message reveal the user_id of a grader to a peer and break anonymization?

Reply:
It will only show non-existent ids. I guess someone could guess a user_id? Should I change the message to not include the id?

Reply:
Re-opening. Will this happen if the grader has left the course?